src/web_voice_processor.ts

```diff
@@ -92,37 +81,85 @@ export class WebVoiceProcessor {
    * `.postMessage({command: 'process', inputFrame: inputFrame})`.
    * @param engine The engine to unsubscribe.
    */
-  public subscribe(engine: PvEngine): void {
+  public static async subscribe(engine: PvEngine): Promise<void> {
```

Comment:
I wonder if it would be better to allow for a single engine **or** an array of engines. I'm thinking of a situation where someone calls `reset()` and then needs to resubscribe all of the engines again. It would be cleaner if they could subscribe their engines all at once.